src/main/java/roomescape/reservation/ReservationJdbcRepository.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.stereotype.Repository;
+import roomescape.reservationtime.ReservationTime;
+import roomescape.theme.Theme;
+
+@Repository
+public class ReservationJdbcRepository implements ReservationRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final SimpleJdbcInsert simpleJdbcInsert;
+
+    public ReservationJdbcRepository(
+            @Autowired final JdbcTemplate jdbcTemplate
+    ) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.simpleJdbcInsert = new SimpleJdbcInsert(jdbcTemplate)
+                .withTableName("reservation")
+                .usingGeneratedKeyColumns("id");
+    }
+
+    @Override
+    public Long save(
+            final Reservation reservation,
+            final Long reservationTimeId,
+            final Long themeId
+    ) {
+        final Map<String, Object> reservationParameter = Map.of(
+                "name", reservation.getName(),
+                "date", reservation.getDate(),
+                "time_id", reservationTimeId,
+                "theme_id", themeId
+        );
+        return simpleJdbcInsert
+                .executeAndReturnKey(reservationParameter)
+                .longValue();
+    }
+
+    @Override
+    public List<Reservation> findAll() {
+        final String sql = "SELECT r.id AS reservation_id, " +
+                "       r.name, " +
+                "       r.date, " +
+                "       time.id AS time_id, " +
+                "       time.start_at AS time_value, " +
+                "       theme.id AS theme_id, " +
+                "       theme.name AS theme_name, " +
+                "       theme.description AS theme_description, " +
+                "       theme.thumbnail AS theme_thumbnail " +
+                "FROM reservation AS r INNER JOIN reservation_time AS time " +
+                "    ON r.time_id = time.id " +
+                "    INNER JOIN theme AS theme " +
+                "    ON r.theme_id = theme.id ";
+        return jdbcTemplate.query(sql, getRowMapper());
+    }
+
+    @Override
+    public Reservation findById(Long id) {
+        final String sql = "SELECT r.id AS reservation_id, " +
+                "       r.name, " +
+                "       r.date, " +
+                "       time.id AS time_id, " +
+                "       time.start_at AS time_value, " +
+                "       theme.id AS theme_id, " +
+                "       theme.name AS theme_name, " +
+                "       theme.description AS theme_description, " +
+                "       theme.thumbnail AS theme_thumbnail " +
+                "FROM reservation AS r " +
+                "    INNER JOIN reservation_time AS time " +
+                "    ON r.time_id = time.id " +
+                "    INNER JOIN theme AS theme " +
+                "    ON r.theme_id = theme.id " +
+                "WHERE r.id=? ";
+        return jdbcTemplate.queryForObject(sql,getRowMapper(), id);
```

Comment:
자동 정렬은 습관적으로 진행해주세요. 😃
매번 의식적으로 정렬하시는게 어려우시다면, IDE 설정에서 자동으로 정렬을 수행해주는 설정을 켤수도 있으니 적용해보셔도 좋을 것 같습니다. 

```suggestion
        return jdbcTemplate.queryForObject(sql, getRowMapper(), id);
```

Reply:
감사합니다.. !!